evm_loader/program/src/entrypoint.rs

```diff
@@ -481,29 +491,35 @@ fn do_partial_call<'a>(
     account_storage: &ProgramAccountStorage,
     accounts: &'a [AccountInfo<'a>],
     instruction_data: Vec<u8>,
+    gas_limit: u64,
 ) -> ProgramResult
 {
     debug_print!("do_partial_call");
 
     let backend = SolanaBackend::new(account_storage, Some(accounts));
     debug_print!("  backend initialized");
 
-    let executor_state = ExecutorState::new(ExecutorSubstate::new(), backend);
+    let executor_state = ExecutorState::new(ExecutorSubstate::new(gas_limit), backend);
     let mut executor = Machine::new(executor_state);
 
     debug_print!("Executor initialized");
 
     debug_print!("   caller: {}", account_storage.origin());
     debug_print!(" contract: {}", account_storage.contract());
 
-    executor.call_begin(account_storage.origin(), account_storage.contract(), instruction_data, u64::max_value());
+    executor.call_begin(
+        account_storage.origin(),
+        account_storage.contract(),
+        instruction_data,
+        gas_limit,
+    ).map_err(|_| ProgramError::InvalidInstructionData)?;
```

Comment:
`map_err` is redundant
`call_begin` now returns Result<(), ProgramError>

Reply:
Exactly! Fixed in both places.